src/ensembl/src/content/app/custom-download/components/checkbox-with-radios/CheckboxWithRadios.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import Checkbox from 'src/shared/checkbox/Checkbox';
+import Radio, { radioOptions } from 'src/shared/radio/Radio';
+
+import styles from './CheckboxWithRadios.scss';
+
+type Props = {
+  radioOptions: radioOptions;
+  label: string;
+  selectedOption: string;
+  onChange: (selectedOption: string) => void;
+};
+
+const CheckboxWithRadios = (props: Props) => {
+  const [checkedStatus, setCheckedStatus] = useState(false);
+
+  useEffect(() => {
+    if (props.selectedOption) {
+      setCheckedStatus(true);
+    }
+  }, []);
+
+  const handleCheckboxOnChange = useCallback(
+    (checkedStatus: boolean) => {
+      setCheckedStatus(checkedStatus);
+      props.onChange('');
+    },
+    [props.selectedOption]
+  );
+
+  const handleOnChange = useCallback(
+    (selectedOption: string) => {
+      props.onChange(selectedOption);
+    },
+    [props.selectedOption]
+  );
```

Comment:
Not sure what's going on with the three functions above:

1) for the first function (useEffect) is it just the same as if we wrote, when we instantiated the state:
```
const [checkedStatus, setCheckedStatus] = useState(Boolean(props.selectedOption));
```
?
(also, perhaps change checkedStatus to isChecked? isChecked definitely reads like a true/false value, while checkedStatus reads as something more complicated)

2) for the third function, is it the same as just passing `props.onChange` to the Radio component as the onChange prop?

3) As for the second function, what's the practical purpose of the `useCallback` hook? Why is it not just a function that sets state and calls a prop function?

Reply:
I've rectified all the above three issues.